src/js/tools/pen.js

```diff
@@ -121,5 +124,20 @@ define(function (require, exports, module) {
     };
     util.inherits(PenTool, Tool);
 
+    /**
+     * Handler for keydown events, installed when the tool is active.
+     *
+     * @param {CustomEvent} event
+     */
+    PenTool.prototype.onKeyDown = function (event) {
+        var flux = this.getFlux(),
+            toolStore = flux.store("tool"),
+            detail = event.detail;
+  
+        if (toolStore.getVectorMode() && detail.keyCode === OS.eventKeyCode.ESCAPE) {
```

Comment:
There could be a reason I'm not thinking of, but wouldn't it be simpler to install an `ESCAPE` shortcut when selecting the pen tool and to remove it when deselecting?